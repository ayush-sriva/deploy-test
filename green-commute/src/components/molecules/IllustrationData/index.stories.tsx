import IllustrationData from '.';
import { ComponentMeta, ComponentStory } from '@storybook/react';

export default {
  title: 'Components/Molecules/IllustrationData',
  component: IllustrationData,
} as ComponentMeta<typeof IllustrationData>;

const IllustrationDataTemplate: ComponentStory<typeof IllustrationData> = (args: any) => (
  <div style={{ background: 'linear-gradient(155.94deg, #EFFEFF 6.2%, #E9FFF4 52.61%)', padding: '30px' }}>
    {' '}
    <IllustrationData {...args} />
  </div>
);

export const IllustrationData1 = IllustrationDataTemplate.bind({});
IllustrationData1.args = {
  url: '/assets/illustrations/illustration2.png',
  children: 'Enter Location to know Time Air Quality index (AQI)',
};

export const IllustrationData2 = IllustrationDataTemplate.bind({});
IllustrationData2.args = {
  url: '/assets/illustrations/illustration4.png',
  children: 'Enter Location to know Time Air Quality (AQI)',
};

export const IllustrationData3 = IllustrationDataTemplate.bind({});
IllustrationData3.args = {
  url: '/assets/illustrations/illustration3.png',
  children: 'Enter your Skills to know how many jobs are in this Location',
};
