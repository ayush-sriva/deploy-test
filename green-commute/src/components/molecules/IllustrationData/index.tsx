/**
 * @author Asish Kalintha <asish.sugun@zemosolabs.com>
 */
import { Grid } from '@mui/material';
import { makeStyles } from '@mui/styles';
import Typography from '../../atoms/Typography/index';
import Illustration from '../../atoms/Illustration';
import responsiveTheme from '../../../theme';
import { ReactNode } from 'react';

interface IProps {
  children: ReactNode;
  url: string;
}

const useStyles = makeStyles({
  container: {
    maxWidth: '417px',
    height: '348px',
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    color: `${responsiveTheme.palette.secondary.dark}`,
    textAlign: 'center',
    marginTop: '40px',
  },
});

const IllustrationData:React.FC<IProps> = ({url, children}) => {
    const classes = useStyles();
  return (
    <div data-testid='illustrationData'>
      <Grid
        container
        direction='column'
        className={classes.container}
        wrap='nowrap'
       
      >
        <Grid item>
          <Illustration url={url} />
        </Grid>

        <Grid item>
          <Typography
            variant='h2'
            className={classes.text}
            
          >
            {children}
          </Typography>
        </Grid>
      </Grid>
    </div>
  );
};

export default IllustrationData;
