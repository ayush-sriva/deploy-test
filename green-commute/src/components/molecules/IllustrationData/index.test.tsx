import renderer from 'react-test-renderer';
import IllustrationData from '.';

describe('IllustrationData', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<IllustrationData children='descreption' url='/assets/illustrations/illustration3.png' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
