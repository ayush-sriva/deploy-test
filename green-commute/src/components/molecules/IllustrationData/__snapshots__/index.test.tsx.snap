// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IllustrationData renders correctly 1`] = `
<div
  data-testid="illustrationData"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap makeStyles-container-1 css-19kb62l-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <img
        alt="illustration"
        src="/assets/illustrations/illustration3.png"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h2 makeStyles-text-2 css-1sra7t5-MuiTypography-root"
      >
        descreption
      </h2>
    </div>
  </div>
</div>
`;
